src/main/java/com/metamx/emitter/core/ParametrizedUriEmitter.java

```diff
@@ -120,10 +121,30 @@ public void close() throws IOException
   @Override
   public void flush() throws IOException
   {
-    synchronized (emitters) {
-      for (Emitter emitter : emitters.values()) {
-        emitter.flush();
+    Exception thrown = null;
+    for (HttpPostEmitter httpPostEmitter : emitters.values()) {
+      try {
+        httpPostEmitter.flush();
       }
+      catch (Exception e) {
+        // If flush was interrupted, exit the loop
+        if (Thread.currentThread().isInterrupted()) {
+          if (thrown != null) {
+            e.addSuppressed(thrown);
+          }
+          throw Throwables.propagate(e);
```

Comment:
`Throwables.propagate` is deprecated in guava now. Maybe it's better not using it here? https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate